### PR TITLE
Build a static dependency environment from core

### PR DESCRIPTION
**Pull request by nmuesch**
### What does this PR do?

Uses pip-compile to generate the list of dependencies required by integrations core using the integration core agent_requirements.in file. 

### Motivation

Keep a static environment of dependencies and keep things up to date with the changes in integrations-core

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: omnibus/config/software/datadog-agent-integrations.rb [hunk 1]
```diff
@@ -83,47 +80,73 @@
...
 
-    # Set frozen requirements
-    pip "freeze > #{install_dir}/agent_requirements.txt"
-
     # Windows pip workaround to support globs
     python_bin = "\"#{windows_safe_path(install_dir)}\\embedded\\python.exe\""
-    python_pip = "pip install -c #{windows_safe_path(install_dir)}\\agent_requirements.txt #{windows_safe_path(project_dir)}"
-    python_pip_req = "pip install -c #{windows_safe_path(install_dir)}\\agent_requirements.txt -r #{windows_safe_path(project_dir)}"
+    python_pip_no_deps = "pip install -c #{windows_safe_path(install_dir)}\\constraint_requirements.txt --no-deps #{windows_safe_path(project_dir)}"
+    python_pip_req = "pip install -c #{windows_safe_path(install_dir)}\\constraint_requirements.txt --require-hashes -r #{windows_safe_path(project_dir)}"
```
Comment (truthbk) [inline omnibus/config/software/datadog-agent-integrations.rb]: As I mentioned in the other PR I think I would rather keep the trailing `#{windows_safe_path(project_dir)}` outside of the string and add it to the commands we call `python_pip_req` from, it's more explicit.

File: omnibus/config/software/datadog-agent-integrations.rb [hunk 2]
```diff
@@ -83,47 +80,73 @@
...
-    pinned_requirements = "#{project_dir}/datadog_checks_base/requirements.in"
     if windows?
-      command("#{python_bin} -m #{python_pip}\\datadog_checks_base")
-      if File.exist? pinned_requirements
-        command("#{python_bin} -m #{python_pip_req}\\datadog_checks_base\\requirements.in")
+      command("#{python_bin} -m #{python_pip_no_deps}\\datadog_checks_base")
+      command("#{python_bin} -m piptools compile --generate-hashes --output-file #{windows_safe_path(project_dir)}\\static_requirements.txt #{windows_safe_path(project_dir)}\\datadog_checks_base\\datadog_checks\\data\\agent_requirements.in")
+
+      # Uninstall the deps that pip-compile installs so we don't include them in the final artifact
+      for dep in UNINSTALL_PIPTOOLS_DEPS
```
Comment (truthbk) [inline omnibus/config/software/datadog-agent-integrations.rb]: Let's do this the way it's done in A5, where we check `UNINSTALL_PIPTOOLS_DEPS` were not part of the agent constraints. It's just safer, we never know what other software definitions might do.

**Review by truthbk — CHANGES_REQUESTED** [review]
Some minor changes for consistency/safety.

File: omnibus/config/software/datadog-agent-integrations.rb (hunk 2 above)
Comment (nmuesch) [reply omnibus/config/software/datadog-agent-integrations.rb]: There are some weird issues with Ruby in terms of doing a File.read on the constraints file to make sure those dependencies aren't there.

**Review by derekwbrown — APPROVED** [review]

**Pull request merged**